configurationproperties-annotationprocessor/src/main/java/com/vaadin/flow/spring/ConfigurationPropertiesAnnotationProcessor.java

```diff
@@ -0,0 +1,386 @@
+package com.vaadin.flow.spring;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.util.Elements;
+import javax.tools.Diagnostic;
+import javax.tools.FileObject;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardLocation;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.ImportDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+import com.github.javaparser.utils.Utils;
+
+@SupportedAnnotationTypes("com.vaadin.flow.spring.SpringConfigurationPropertiesGenerator")
+public class ConfigurationPropertiesAnnotationProcessor extends AbstractProcessor
+{
+
+    private static final String INITPARAMETERS_CLASS_NAME = "com.vaadin.flow.server.InitParameters";
+    private Messager messager;
+    private Map<String, CompilationUnit> compilationUnitCache = new HashMap<>();
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
```

Comment:
the method is already defined as `synchronized` in the superclass

Reply:
So why do you need to duplicate it here ?

`synchronized` in the super class doesn't mean that you have to mark the overriden method .
`synchronized` in the superclass protects superclass state for some reasons. And it will protect it regardless of whether you  repeat it here or not : the super method is called .

So I didn't get the answer.